src/mame/audio/exidy.cpp

```diff
@@ -788,9 +790,41 @@ DEFINE_DEVICE_TYPE(EXIDY_MTRAP, mtrap_sound_device, "mtrap_sound", "Exidy SFX+PS
 
 mtrap_sound_device::mtrap_sound_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
 	: venture_sound_device(mconfig, EXIDY_MTRAP, tag, owner, clock)
+	, m_cvsd_clk(false)
 {
 }
 
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void mtrap_sound_device::device_start()
+{
+	common_sh_start();
+
+	/* 8253 */
+	m_freq_to_step = (1 << 24) / SH8253_CLOCK;
+
+	sh8253_register_state_globals();
+	/* timer */
+	m_cvsd_timer = timer_alloc(TIMER_CVSD);
+	m_cvsd_timer->adjust(attotime::from_hz(CVSD_CLOCK*2.0));
+
+	save_item(NAME(m_cvsd_clk));
+}
+
+void mtrap_sound_device::device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr)
+{
+	switch(id)
+	{
+		case TIMER_CVSD:
+			m_cvsd_clk = !m_cvsd_clk;
+			m_cvsd->clock_w(m_cvsd_clk);
+			m_cvsd_timer->adjust(attotime::from_hz(CVSD_CLOCK*2.0));
+		break;
+	}
+}
```

Comment:
There’s a `clock_device` (src/devices/machine/clock.h) that generates a symmetric clock.  Even if that didn’t exist, you could use a periodic timer to do this more cleanly.

Reply:
This has been switched to using a periodic timer